Source/core/TokenizedStringList.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+#include "Module.h"
+
+namespace WPEFramework {
+namespace Core {
+
+    template <char SEPARATOR, bool REMOVE_WHITESPACES>
+    class TokenizedStringList {
+    public:
+        TokenizedStringList() = default;
+        ~TokenizedStringList() = default;
+        TokenizedStringList(const TokenizedStringList&) = default;
+        TokenizedStringList& operator=(const TokenizedStringList&) = default;
+
+        explicit TokenizedStringList(const string& buffer)
+        {
+            Tokenize(buffer);
+        }
+
+        void Tokenize(const string& buffer)
+        {
+
+            if (!buffer.empty()) {
+                //remove spaces and tokenize by comma
+                if (REMOVE_WHITESPACES) {
+                    string result = buffer;
+                    result.erase(std::remove_if(result.begin(), result.end(), ::isspace), result.end());
+                    std::istringstream iss(result);
+                }
+                std::istringstream iss(buffer);
+
+                string substring;
+                while (iss.good()) {
+                    getline(iss, substring, SEPARATOR);
+                    _parts.push_back(substring);
+                }
+            }
+        }
+
+        void Clear()
+        {
+            _parts.clear();
+        }
+
+        const string& First() const
+        {
+            return _parts.front();
+        }
+
+        string All() const
+        {
+            string result;
+
+            if (!_parts.empty()) {
+
+                for (const auto& part : _parts) {
+                    result += part;
+                    result += SEPARATOR;
+                }
+
+                result.erase(result.rfind(','));
```

Comment:
I am afraid rgat this should read result.erase(result.rfind(SEPARATOR));  :-)
And could you add the copyright header to this file (C) metroogical blah blah from another file ..